Arrowgene.Ddon.Database/Sql/Core/Migration/00000010_BinaryDataMigration.cs

```diff
@@ -0,0 +1,336 @@
+using Arrowgene.Ddon.Shared.Entity;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Logging;
+using System.Data.Common;
+
+namespace Arrowgene.Ddon.Database.Sql.Core.Migration
+{
+    public class BinaryDataMigration : IMigrationStrategy
+    {
+        private static readonly ILogger Logger = LogProvider.Logger<Logger>(typeof(BinaryDataMigration));
+
+        public uint From => 9;
+        public uint To => 10;
+
+        private readonly DatabaseSetting DatabaseSetting;
+
+        public BinaryDataMigration(DatabaseSetting databaseSetting)
+        {
+            DatabaseSetting = databaseSetting;
+        }
+
+        public bool Migrate(IDatabase db, DbConnection conn)
+        {
+            string adaptedSchema = DdonDatabaseBuilder.GetAdaptedSchema(DatabaseSetting, "Script/binarydata_migration.sql");
+            db.Execute(conn, adaptedSchema);
+
+            S2CCharacterDecideCharacterIdRes pcap = EntitySerializer.Get<S2CCharacterDecideCharacterIdRes>().Read(data_Dump_13);
+
+            db.ExecuteReader(conn, @"SELECT ddon_character.character_id FROM ddon_character;", action => { }, reader =>
+            {
+                while (reader.Read())
+                {
+                    uint characterId = db.GetUInt32(reader, "character_id");
+                    Logger.Info($"Assigning pcap binary data to character ID {characterId}");
+                    db.ExecuteNonQuery(conn, @"INSERT INTO ddon_binary_data(character_id, binary_data)
+                                                        VALUES(@character_id, @binary_data);",
+                            action =>
+                            {
+                                db.AddParameter(action, "character_id", characterId);
+                                db.AddParameter(action, "binary_data", pcap.BinaryData);
```

Comment:
We are we migrating pcap data?

Reply:
I see but considering we don't have any tutorials today, that means player will get many popups when they login with newer characters now?